diff --git a/target/product/aosp_arm64.mk b/target/product/aosp_arm64.mk
index 5052404..6e6aae3 100644
--- a/target/product/aosp_arm64.mk
+++ b/target/product/aosp_arm64.mk
@@ -78,3 +78,8 @@ PRODUCT_BRAND := Android
 PRODUCT_MODEL := AOSP on ARM64
 
 PRODUCT_NO_BIONIC_PAGE_SIZE_MACRO := true
+
+# K11C factory vendor optional HAL declarations.
+ifeq (aosp_arm64,$(TARGET_PRODUCT))
+PRODUCT_PACKAGES += k11c_framework_compatibility_matrix
+endif
diff --git a/target/product/gsi/Android.bp b/target/product/gsi/Android.bp
index 3fa6e76..0d33394 100644
--- a/target/product/gsi/Android.bp
+++ b/target/product/gsi/Android.bp
@@ -178,6 +178,9 @@ android_system_image {
     name: "aosp_system_image",
     defaults: ["android_gsi_defaults"],
     deps: [
+        // K11C factory vendor optional HAL declarations.
+        "k11c_framework_compatibility_matrix",
+
         // handheld_system_ext
         "AccessibilityMenu",
         "WallpaperCropper",
