diff --git a/target/product/aosp_arm64.mk b/target/product/aosp_arm64.mk
--- a/target/product/aosp_arm64.mk
+++ b/target/product/aosp_arm64.mk
@@ -23,6 +23,7 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/generic_system.mk)
 # Enable mainline checking for excat this product name
 ifeq (aosp_arm64,$(TARGET_PRODUCT))
 PRODUCT_ENFORCE_ARTIFACT_PATH_REQUIREMENTS := relaxed
+PRODUCT_PACKAGES += k11c-ril-compat
 endif
 
 #
diff --git a/target/product/gsi/Android.bp b/target/product/gsi/Android.bp
--- a/target/product/gsi/Android.bp
+++ b/target/product/gsi/Android.bp
@@ -210,2 +210,3 @@
         "ThemePicker",
+        "k11c-ril-compat",
     ] + select(product_variable("debuggable"), {
