diff --git a/target/product/gsi/Android.bp b/target/product/gsi/Android.bp
--- a/target/product/gsi/Android.bp
+++ b/target/product/gsi/Android.bp
@@ -209,6 +209,7 @@
         true: ["frameworks-base-overlays-debug"],
         default: [],
     }),
+    file_contexts: ":k11c_gsi_file_contexts",
     enabled: select(soong_config_variable("gsi", "building_gsi"), {
         true: true,
         default: false,
